src/System.Numerics.Tensors/System/Numerics/CompressedSparseTensor.cs

```diff
@@ -385,35 +397,35 @@ public override Tensor<T> Reshape(params int[] dimensions)
 
         public override DenseTensor<T> ToDenseTensor()
         {
-            var denseTensor = new DenseTensor<T>(dimensions, reverseStride: IsReversedStride);
+            var denseTensor = new DenseTensor<T>(Dimensions, reverseStride: IsReversedStride);
 
             var compressedIndex = 0;
 
+            var compressedCountsSpan = compressedCounts.Span;
+            var indicesSpan = indices.Span;
+            var valuesSpan = values.Span;
             for (int valueIndex = 0; valueIndex < nonZeroCount; valueIndex++)
             {
-                while (valueIndex >= compressedCounts[compressedIndex + 1])
+                while (valueIndex >= compressedCountsSpan[compressedIndex + 1])
                 {
                     compressedIndex++;
                     Debug.Assert(compressedIndex < compressedCounts.Length);
                 }
 
-                var index = indices[valueIndex] + compressedIndex * strides[compressedDimension];
+                var index = indicesSpan[valueIndex] + compressedIndex * strides[compressedDimension];
 
-                denseTensor.SetValue(index, values[valueIndex]);
+                denseTensor.SetValue(index, valuesSpan[valueIndex]);
             }
 
             return denseTensor;
         }
 
         public override CompressedSparseTensor<T> ToCompressedSparseTensor()
         {
-            var newValues = new T[NonZeroCount];
-            Array.Copy(values, newValues, NonZeroCount);
-
-            var newIndicies = new int[NonZeroCount];
-            Array.Copy(indices, newIndicies, NonZeroCount);
+            var newValues = values.Span.Slice(0, nonZeroCount).ToArray();
```

Comment:
Re-write as follows?
```C#
var newValues = values.Slice(0, nonZeroCount).ToArray();
```

Reply:
Slicing of Span is cheaper than slicing of Memory. But it does not matter much here because `ToArray()` will dominate the cost in either case.